src/frontend/src/optimizer/backfill_order_strategy.rs

```diff
@@ -364,11 +364,43 @@ mod common {
     }
 }
 
-// FIXME(kwannoel): we can flatten the strategy earlier
-/// We don't use query binder directly because its binding rules are different from a query.
-/// We only bind tables, materialized views and sources.
-/// Queries won't bind duplicate relations in the same query context.
-/// But backfill order strategy can have duplicate relations.
+pub mod display {
+    use risingwave_common::catalog::ObjectId;
+    use risingwave_pb::stream_plan::BackfillOrder;
+
+    use crate::session::SessionImpl;
+
+    fn get_table_name(session: &SessionImpl, id: ObjectId) -> crate::error::Result<String> {
+        let catalog_reader = session.env().catalog_reader().read_guard();
+        let table_catalog = catalog_reader.get_any_table_by_id(&(id.into()))?;
+        let table_name = table_catalog.name();
+        let db_id = table_catalog.database_id;
+        let schema_id = table_catalog.schema_id;
+        let schema_catalog = catalog_reader.get_schema_by_id(&db_id, &schema_id)?;
+        let schema_name = schema_catalog.name();
+        let name = format!("{}.{}", schema_name, table_name);
+        Ok(name)
+    }
+
+    pub(super) fn print_backfill_order_in_dot_format(
+        session: &SessionImpl,
+        order: BackfillOrder,
+    ) -> crate::error::Result<String> {
+        let mut result = String::new();
+        result.push_str("digraph G {\n");
```

Comment:
I intentionally do not use a rust crate for graphviz. We require less LoC without it, and the structure here is rather clear.
So there's no need to add more abstractions.